Initial commit of heims data dict validator

HughParsonage opened the pull request:
The role of the data dictionary at this version should be to:

1. Recode the names (E306 etc) into human-readable names
2. Check that all values in the variables comply with the data dictionary coding requirements. 
3. For a given variable, identify values that merely label the observation as missing, rather than having that particular value. 

QC tasks:

Go the following link and select 3 variables.

http://heimshelp.education.gov.au/sites/heimshelp/dictionary/pages/data-element-dictionary `> Elements by number`

Then make sure, by writing some unit tests, that for these variables the data dictionary performs to spec.

Reviewing the source code of this package is optional. It should comply with https://github.com/grattaninstitute/R-Style-Guide

HughParsonage committed:
Initial commit of heims data dict validator

HughParsonage committed:
CRAN NOTE avoidance

HughParsonage committed:
Add E490 decoder

HughParsonage committed:
Provide .progress_cat arg to validate_elements

HughParsonage committed:
Initial commit of renamer

HughParsonage committed:
Ensure returns length-1 result

HughParsonage committed:
Use non-hard-coded path to decoder

HughParsonage committed:
E490 decoder reproducibility

HughParsonage committed:
Export rename heims

lucilled commented:
Okay, sure. I'm in the middle of something right now but will give it a go within 24 hours :)

HughParsonage committed:
Permit renaming vec elements, not just names(v)

HughParsonage committed:
RM duplicate entry in data_dict

HughParsonage committed:
Import if_else from dplyr

HughParsonage committed:
(Add notin util function)

HughParsonage committed:
Add long_name abbreviations and extra list eles

HughParsonage committed:
Treat element number suffixes when renaming

HughParsonage committed:
(documentation)

HughParsonage committed:
Ignore -A suffix when validating

HughParsonage committed:
Add hard-to-find elements to dict

HughParsonage committed:
Relax element identification

HughParsonage committed:
Add U490

HughParsonage committed:
Add semester suffix to element2name

HughParsonage committed:
Separate suffixes from decoding

HughParsonage committed:
Add `every` function (not crazy about name)

HughParsonage committed:
valid vectorized element of data dictionary

HughParsonage committed:
Found absent closing parenthesis

lucilled commented:
Hey @HughParsonage - I'm just getting to this now. Have cloned heims and pointed RStudio towards it, but I can't work out how to find/open the data dictionary that this pull request is referring to. Am pretty sure I have forgotten a key set up step - any idea which?

HughParsonage committed:
Add valid function

HughParsonage committed:
Change `invalid` to valid

Commit:
Add test

HughParsonage committed:
Add prop_elements_valid

HughParsonage committed:
Add E490 decoder

HughParsonage committed:
Merge branch 'data-dictionary' of https://github.com/HughParsonage/heims into data-dictionary

HughParsonage committed:
Document prop_elements_valid

HughParsonage committed:
Vectorize is.Date; add absent valid()s

HughParsonage committed:
Add count of invalid elements

HughParsonage committed:
Document char argument

HughParsonage committed:
Vectorize valid functions

HughParsonage committed:
Spurious invalidation corrected by inspection

HughParsonage committed:
Add ad_hoc validation steps

HughParsonage committed:
Add E477 ad hoc validation

HughParsonage committed:
Ad hoc validation notes

HughParsonage committed:
Add past elements to E490

HughParsonage committed:
Contemplate E446A

This has a suffix, but it appears have a distinct encoding, viz. "Y", "N", or (missing).

HughParsonage committed:
Add validation notes due to inspection of data

HughParsonage committed:
Validate E312 to accommodate AMC values

HughParsonage committed:
Add valid missing value to valid() command

HughParsonage committed:
Add U to Gender; Prettier documentation

HughParsonage committed:
Correct E347 encoding

HughParsonage committed:
Revise data dictionary due inspection

HughParsonage committed:
Fix prev erroneous entry

HughParsonage committed:
Contemplate mixture of dbl and int

HughParsonage committed:
Add past entries

http://heimshelp.education.gov.au/sites/heimshelp/2010_data_requirements/2010dataelements/pages/573

HughParsonage committed:
E095 decoder (2016)

HughParsonage committed:
Add decoders (incl 1201 by insp for English)

HughParsonage committed:
Add decoders (incl 1201 by insp for English)

HughParsonage committed:
Merge branch 'decoders' of https://github.com/HughParsonage/heims into decoders

HughParsonage committed:
Add decoders

HughParsonage committed:
Merge branch 'decoders' of https://github.com/HughParsonage/heims into decoders

HughParsonage committed:
Add ABS country codes

HughParsonage committed:
Add 346 decoder

HughParsonage committed:
Set validate to presence in decoder, not range

HughParsonage committed:
Validate broader regions than country code

HughParsonage committed:
Rename ABS country codes to be correct

HughParsonage committed:
Add E346 decoder

HughParsonage committed:
Add entries that may be admiteted (need to be treated)

HughParsonage committed:
Allow dates from 1899 (due to ExcelMac issue)

HughParsonage committed:
Expand missing values

HughParsonage committed:
Initial commit of ad_hoc_prepare list el

HughParsonage committed:
Add past country codes as valid codes

HughParsonage committed:
Add Serbia and Montenegro

HughParsonage committed:
Ensure if_else returns int

HughParsonage committed:
Add 562 decoder (FOE / 10)

HughParsonage committed:
heims for if_else etc

HughParsonage committed:
Coalesce 40000 as 49999 etc

HughParsonage committed:
Add if_else to namespace

HughParsonage committed:
Correct E493. Add test.

HughParsonage committed:
Initial commit of decode using

HughParsonage committed:
Rename E306 decoder

HughParsonage committed:
Ad E358 decoder

HughParsonage committed:
Reexpress if_else to avoid manpage warnings

HughParsonage committed:
Requires unlisting

HughParsonage committed:
Add E562 decoder

HughParsonage committed:
Add decode_using method

HughParsonage committed:
Validate TER test

HughParsonage committed:
Add E461 decoder

HughParsonage committed:
Student id needs to rm leading zeroes

HughParsonage committed:
Update documentation

HughParsonage committed:
Add data

HughParsonage committed:
Add 358 decoder

HughParsonage committed:
Put decoders

HughParsonage committed:
Put FOE Ittima decoder

HughParsonage committed:
Add browse_elements

HughParsonage committed:
Initial commit of decode_heims

HughParsonage committed:
Add as.integer64 to utils

HughParsonage committed:
Add decoders doc

HughParsonage committed:
Don't use Country_code in decoding table

HughParsonage committed:
Decoders require keys

HughParsonage committed:
Date of birth decoder

HughParsonage committed:
Add E358 decoder (disability) to dict

HughParsonage committed:
Add post_fst remark as .fst doesn't preserve 64

HughParsonage committed:
More consistent decoder in dict

HughParsonage committed:
Add E477 and E459 decoders

HughParsonage committed:
Add E330 decoder

HughParsonage committed:
Correct path to E330 decoder

HughParsonage committed:
Add E922 decoder

HughParsonage committed:
Add E392 decoder

HughParsonage committed:
Add E311_decoder

HughParsonage committed:
Add E392_decoder to dict

HughParsonage committed:
Add E331_decoder

HughParsonage committed:
Add E089_decoder

HughParsonage committed:
Add E463_decoder

HughParsonage committed:
Add 312_decoder to dict

HughParsonage committed:
Shorten E358_decoded

HughParsonage committed:
Narrower decoded columns

HughParsonage committed:
Simplify (uniqueN small as expected)

HughParsonage committed:
Nicer country decoding, na.rm = TRUE

HughParsonage committed:
Add Yugoslavia

HughParsonage committed:
Neaten widht

HughParsonage committed:
Add documentation

HughParsonage committed:
Decode and rename decode_heims

HughParsonage committed:
The names need to be present before ordered

HughParsonage committed:
Add E355 decoder

HughParsonage committed:
Add and abbreviate E490 decoder

Original decoding would have resulted in very wide columns, so have just used whether or not they have deferred or not.

HughParsonage committed:
Add E316 decoder to dict

HughParsonage committed:
Test course and load 2015 files

HughParsonage committed:
Add E464 decoder (copy of FOE decoder)

HughParsonage committed:
Add E464 doc

HughParsonage committed:
Add U490 decoder

HughParsonage committed:
Add E551 decoder

HughParsonage committed:
`post_fst` in dict now a DT-function

HughParsonage committed:
Add check for multiple entries in dict

HughParsonage committed:
Convert dates to Dates

HughParsonage committed:
New long name due to inspection of course2015

HughParsonage committed:
New validate CHESSN

HughParsonage committed:
Decode E446A

HughParsonage committed:
Add show_progress arg to decode_heims

So that I don't get impatient?

HughParsonage committed:
Permit younglings

HughParsonage committed:
Sys.time() is a function, not an object

HughParsonage committed:
Permit younglings

HughParsonage committed:
Add Africa from 1998 country codes

Permit NAs in validation

HughParsonage committed:
Case-insensitivity in browse_elements

HughParsonage committed:
Rename and decode methods for E455

HughParsonage committed:
ATSI no longer coded

HughParsonage committed:
NA_if not what I thought

HughParsonage committed:
Add `foename` to FOE decoder

HughParsonage committed:
Add E327 decoder

HughParsonage committed:
Export is.integer64, use correct fread NA

HughParsonage committed:
Make Semesters logical

Save space, better semantically

HughParsonage committed:
Clarify semantics of immigration variables

HughParsonage committed:
Express the point of E465

HughParsonage committed:
Mark missing value

HughParsonage committed:
Add changes to dict, add utilities function

HughParsonage committed:
Prettier progress output

HughParsonage committed:
Update documentation

HughParsonage committed:
Use fastmatch here

HughParsonage committed:
Put do_ad_hoc_prepares inside decode heims

HughParsonage committed:
Uppercase E (course uses e550)

HughParsonage committed:
Initial commit of fread_heims

HughParsonage committed:
Add test for decoded values

HughParsonage committed:
Use lubridate::ymd (4x speedup)

HughParsonage committed:
Add lubridate's ymd to namespace

HughParsonage committed:
(typo)

HughParsonage committed:
Rerun put_decoders.R

HughParsonage committed:
Merge pull request #2 from HughParsonage/decoders

Decoders

HughParsonage committed:
Add E337 decoder

http://heimshelp.education.gov.au/sites/heimshelp/2016_data_requirements/2016dataelements/pages/337

HughParsonage committed:
Add E337 doc

HughParsonage committed:
Add load decoding test

HughParsonage commented:
Provisionally approved
